pkg/kubelet/dockertools/manager.go

```diff
@@ -807,7 +807,7 @@ func (dm *DockerManager) podInfraContainerChanged(pod *api.Pod, podInfraContaine
 			glog.V(4).Infof("host: %v, %v", pod.Spec.SecurityContext.HostNetwork, networkMode)
 			return true, nil
 		}
-	} else {
+	} else if dm.networkPlugin.Name() != "cni" || dm.networkPlugin.Name() != "kubenet" {
```

Comment:
Argh, Fix for this coming shortly, copy paste error due to faulty rebase of upstream and rebuilding my build env. It should be &&